block/pending_blocks_test.go

```diff
@@ -1,74 +1,90 @@
 package block
 
 import (
+	"context"
 	"sort"
 	"testing"
 
+	test "github.com/rollkit/rollkit/test/log"
+
+	"github.com/rollkit/rollkit/store"
+
 	"github.com/stretchr/testify/require"
 
 	"github.com/rollkit/rollkit/types"
 )
 
 func TestGetPendingBlocks(t *testing.T) {
 	require := require.New(t)
-	pb := NewPendingBlocks()
+	ctx := context.Background()
+	pb := newPendingBlocks(t)
 	for i := uint64(0); i < 5; i++ {
-		pb.addPendingBlock(types.GetRandomBlock(i, 0))
+		require.NoError(pb.store.SaveBlock(ctx, types.GetRandomBlock(i, 0), &types.Commit{}))
+		pb.store.SetHeight(ctx, i)
 	}
-	blocks := pb.getPendingBlocks()
+	blocks, _ := pb.getPendingBlocks(ctx)
 	require.True(sort.SliceIsSorted(blocks, func(i, j int) bool {
 		return blocks[i].Height() < blocks[j].Height()
 	}))
 }
 
 func TestRemoveSubmittedBlocks(t *testing.T) {
 	require := require.New(t)
-	pb := NewPendingBlocks()
-	for i := uint64(0); i < 5; i++ {
-		pb.addPendingBlock(types.GetRandomBlock(i, 0))
+	ctx := context.Background()
+	pb := newPendingBlocks(t)
+	const nBlocks = 5
+	for i := uint64(1); i <= nBlocks; i++ {
+		require.NoError(pb.store.SaveBlock(ctx, types.GetRandomBlock(i, 0), &types.Commit{}))
+		pb.store.SetHeight(ctx, i)
 	}
-	blocks := pb.getPendingBlocks()
-	pb.removeSubmittedBlocks(blocks)
+	pb.setLastSubmittedHeight(nBlocks)
 	require.True(pb.isEmpty())
 }
 
 func TestRemoveSubsetOfBlocks(t *testing.T) {
 	require := require.New(t)
-	pb := NewPendingBlocks()
-	for i := uint64(0); i < 5; i++ {
-		pb.addPendingBlock(types.GetRandomBlock(i, 0))
+	ctx := context.Background()
+	pb := newPendingBlocks(t)
+	for i := uint64(1); i <= 5; i++ {
+		require.NoError(pb.store.SaveBlock(ctx, types.GetRandomBlock(i, 0), &types.Commit{}))
+		pb.store.SetHeight(ctx, i)
 	}
 	// Remove blocks with height 1 and 2
-	pb.removeSubmittedBlocks([]*types.Block{
-		types.GetRandomBlock(1, 0),
-		types.GetRandomBlock(2, 0),
-	})
-	remainingBlocks := pb.getPendingBlocks()
+	pb.setLastSubmittedHeight(2)
+	remainingBlocks, err := pb.getPendingBlocks(ctx)
+	require.NoError(err)
 	require.Len(remainingBlocks, 3, "There should be 3 blocks remaining")
 	for _, block := range remainingBlocks {
-		require.Contains([]uint64{0, 3, 4}, block.Height(), "Only blocks with height 0, 3, and 4 should remain")
+		require.Contains([]uint64{3, 4, 5}, block.Height(), "Only blocks with height 3, 4 and 5 should remain")
 	}
 }
 
 func TestRemoveAllBlocksAndVerifyEmpty(t *testing.T) {
```

Comment:
Yeah, make sense. I was just refactoring existing code. But because of new assumptions, now the tests are more similar.
I will turn them into tabular tests.